### ISSUE TITLE
Error trying to update uamqp

### ISSUE THREAD

**Issue by pierreca** [Azure/azure-iot-cli-extension#31]
Trying to monitor events on an Azure IoT hub, the freshly installed CLI and IoT extension requests an update to uamqp that fails. not sure what the maps stuff is doing in there too...

`(MacOS X High Sierra: Darwin <hostname> 17.5.0 Darwin Kernel Version 17.5.0: Fri Apr 13 19:32:32 PDT 2018; root:xnu-4570.51.2~1/RELEASE_X86_64 x86_64)`

```sh
Pierres-iMac:~ pierreca$ az iot hub monitor-events --hub-name iot-sdks-test
Dependency update required for IoT extension version: 0.4.5. 
Updated dependency must be compatible with uamqp 0.1.0b5. Continue? (y/n) -> y
Updating required dependency...
Collecting uamqp~=0.1.0b5
  Downloading https://files.pythonhosted.org/packages/dd/12/84c134e8676526ec73c4748ed4ed659743aad828fe9dee150941d495831b/uamqp-0.1.0rc1.tar.gz (5.6MB)
    100% |████████████████████████████████| 5.6MB 64.3MB/s 
Collecting certifi>=2017.4.17 (from uamqp~=0.1.0b5)
  Downloading https://files.pythonhosted.org/packages/7c/e6/92ad559b7192d846975fc916b65f667c7b8c3a32bea7372340bfe9a15fa5/certifi-2018.4.16-py2.py3-none-any.whl (150kB)
    100% |████████████████████████████████| 153kB 48.2MB/s 
azure-cli-maps 0.1.0 requires azure-mgmt-maps==0.1.0, which is not installed.
Installing collected packages: certifi, uamqp
Exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/pip/_internal/basecommand.py", line 228, in main
    status = self.run(options, args)
  File "/usr/local/lib/python3.6/site-packages/pip/_internal/commands/install.py", line 335, in run
    use_user_site=options.use_user_site,
  File "/usr/local/lib/python3.6/site-packages/pip/_internal/req/__init__.py", line 49, in install_given_reqs
    **kwargs
  File "/usr/local/lib/python3.6/site-packages/pip/_internal/req/req_install.py", line 748, in install
    use_user_site=use_user_site, pycompile=pycompile,
  File "/usr/local/lib/python3.6/site-packages/pip/_internal/req/req_install.py", line 961, in move_wheel_files
    warn_script_location=warn_script_location,
  File "/usr/local/lib/python3.6/site-packages/pip/_internal/wheel.py", line 216, in move_wheel_files
    prefix=prefix,
  File "/usr/local/lib/python3.6/site-packages/pip/_internal/locations.py", line 165, in distutils_scheme
    i.finalize_options()
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/distutils/command/install.py", line 248, in finalize_options
    "must supply either home or prefix/exec-prefix -- not both")
distutils.errors.DistutilsOptionError: must supply either home or prefix/exec-prefix -- not both
Failure updating uamqp 0.1.0b5. Aborting...
```

**Comment by digimaun** [comment]
Thanks for reporting this @pierreca. Can you try reinstalling/using the IoT extension on 1 or 2 versions back Azure CLI? Yes the azure mgmt maps output is extremely peculiar.

**Comment by troydai** [comment]
What's the version of the `pip`

**Comment by pierreca** [comment]
```sh
12:58 $ pip --version
pip 10.0.1 from /Library/Python/2.7/site-packages/pip-10.0.1-py2.7.egg/pip (python 2.7)
```

**Comment by pierreca** [comment]
synced offline with @digimaun 
this is a combination of 2 issues: 
1. the pip version installed with homebrew on mac has some issues and a fix is needed in the extension to circumvent that
  - https://github.com/pypa/pip/pull/4103
  - https://github.com/Homebrew/brew/pull/821
  - https://stackoverflow.com/questions/24257803/distutilsoptionerror-must-supply-either-home-or-prefix-exec-prefix-not-both
2. a uamqp wheel for macos was missing - this has been fixed already.

@digimaun please update/close at your leisure when you think the situation is back to normal, thanks for your help :)

**Comment by digimaun** [comment]
This issue is now resolved using at least v0.5.0 of the extension.
